### PR TITLE
Text.pad and Text.trim

### PR DESCRIPTION
**Pull request by radeusgd**
### Pull Request Description

Implements https://www.pivotaltracker.com/story/show/181265516

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - ~~If GUI codebase was changed: Enso GUI was tested when built using BOTH `./run dist` and `./run watch`.~~

### CODE REVIEW

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Extensions.enso [hunk 1]
```diff
@@ -1022,3 +1033,95 @@ Text.to_case case_option=Case.Lower locale=Locale.Default = case case_option of
...
+Text.pad length=0 with_pad=' ' at=Location.End =
+    with_pad_length = with_pad.length
+    if with_pad_length == 0 then Error.throw (Illegal_Argument_Error "`with_pad` must not be an empty string.") else
+        pad_size = length - this.length
+        if pad_size <= 0 then this else
+            full_repetitions = pad_size.div with_pad_length
+            remainder = pad_size % with_pad_length
+            case at of
+                Location.Start ->
+                    with_pad.take (Text_Sub_Range.Last remainder) + with_pad.repeat full_repetitions + this
```
Comment (jdunkerley) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Extensions.enso:1071]: is it worth short-circuiting if remainder == 0? Especially as Last would involve a scan of the with_pad?

File: test/Tests/src/Data/Text_Spec.enso [hunk 2]
```diff
@@ -627,6 +638,72 @@ spec =
...
+            "".pad 0 "" . should_fail_with Illegal_Argument_Error
+
+            "".pad 0 . should_equal ""
+            "ABC".pad 3 . should_equal "ABC"
+            "AB".pad -1 . should_equal "AB"
+            "ABC".pad -100 . should_equal "ABC"
+
+            'a\u{301}'.pad 2 . should_equal 'a\u{301} '
+            "".pad 2 'a\u{302}' . should_equal 'a\u{302}a\u{302}'
+            'XX'.pad 5 'yy\u{301}' . should_equal 'XXyy\u{301}y'
```
Comment (jdunkerley) [inline test/Tests/src/Data/Text_Spec.enso:660]: can we do one more test case where the pad_string is truncated on a mulit-char graphene cluster please

**Review by jdunkerley — CHANGES_REQUESTED** [review]
Couple of very minor ones but otherwise LGTM

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Extensions.enso (hunk 1 above)
Comment (radeusgd) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Extensions.enso:1071]: Made Last avoid the scan.

**Review by jdunkerley — APPROVED** [review]

**Pull request merged**